fuse-ts-shotcut.c

```diff
@@ -140,12 +135,16 @@ int find_cutmarks_in_shotcut_project_file (int *inframe, int *outframe, int *bla
 		debug_printf ("find_cutmarks: no valid XML!\n");
 		return 1;
 	}
+
 	mxml_node_t *node;
 	node = mxmlFindElement (xmldoc, xmldoc, "producer", "id", "producer0", MXML_DESCEND);
 	if (NULL == node) {
-		debug_printf ("find_cutmarks: node with id 'producer0' not found!\n");
-		mxmlRelease (xmldoc);
-		return 2;
+		node = mxmlFindElement (xmldoc, xmldoc, "chain", "id", "chain0", MXML_DESCEND);
```

Comment:
If support for Shotcut 19 is dropped, is the lookup of "producer0" still necessary or does shotcut always write a "chain0" ?

Reply:
Support for Shotcut <22 is not generally dropped; the workflow "copy to computer, edit there, copy back" is still working and has, anecdotally, been used for quite a while. Also, we're still generating`producer0` in the template ourselfs.

If you'd like to make a clean cut, I can rearrange the template and parsing-code to only support the chain element. I did not test Shotcut 22 for Windows or Linux yet, but on MacOS Shotcut 22 always generates the chain Element in our usecase and converts producer to chain.

Reply:
Yes, if basically dropping support for older versions, I'd prefer to include a more recent template to be more future-compatible. Maybe the change is not so huge and it can still be read successfully from older shotcut versions too.

It just seems like a very good opportunity to create a blank project in shotcut, drop a single file in, save that project file and strip it down to the minimum.

Reply:
Alright, I agree. I'll update the Template tomorrow and also verify what Shotcut 19 will do with it.

Reply:
@kunsei argued, that debian stable, which is widely used at C3VOC, still has 21.01.29, so that should be the baseline. I'll verify compatibility with 21.01.29, too.

Reply:
Shotcut 21.01 will work fine with the Tempfile-Readonly-Solution but still saves as the `producer0`style Elements and can't open a File with `chain0` style Elements. I would therefore upgrade the Template to what Shotcut 21 expects and support 21+ without bigger changes.

Reply:
Only difference for Shotcut 21 was the specified mlt version, which I upgraded. The amount of "legacy" code is minute (only the lookup of the `producer0` element – even the parsing code is the same and shared, so that I don't think we should drop Shotcut 21 support just for the sake of removing ~6 lines.

The latest Shotcut 21 was released end of Dec 2021; Shotcut 22 is officially released since Jan 2022.